tsdb/head_test.go

```diff
@@ -1988,6 +1988,74 @@ func TestHeadLabelValuesWithMatchers(t *testing.T) {
 	}
 }
 
+func TestHeadLabelNamesWithMatchers(t *testing.T) {
+	head, _ := newTestHead(t, 1000, false)
+	defer func() {
```

Comment:
Same as the other comment, `t.Cleanup` instead of defer.

Reply:
Done: https://github.com/prometheus/prometheus/pull/9083/commits/8bf71bd875bc4f3cfdf1497bf10be56e73468343